ambry-server/src/main/java/com/github/ambry/server/AmbryRequests.java

```diff
@@ -693,6 +696,37 @@ public void handleReplicaMetadataRequest(NetworkRequest request) throws IOExcept
             logger.trace("{} Time used to find entry since: {}", partitionId,
                 (SystemTime.getInstance().milliseconds() - partitionStartTimeInMs));
 
+            // Compute CRC for backup recovery and verification
+            if (replicaMetadataRequestInfo.getReplicaPath().startsWith(BackupCheckerThread.DR_Verifier_Keyword)) {
+              EnumSet<StoreGetOptions> storeGetOptions = EnumSet.of(StoreGetOptions.Store_Include_Deleted,
+                  StoreGetOptions.Store_Include_Expired);
+              List<MessageInfo> newMessageInfos = new ArrayList<>();
+              // for-each blob
+              findInfo.getMessageEntries().forEach(minfo -> {
+                MessageInfo newMsgInfo = minfo;
+                MessageReadSet rdset = null;
+                // get crc of entire blob including blob-properties, user-md, content etc.
+                try {
+                  List<StoreKey> keys = getConvertedStoreKeys(Collections.singletonList(minfo.getStoreKey()));
+                  rdset = store.get(keys, storeGetOptions).getMessageReadSet();
+                  rdset.doPrefetch(0, minfo.getSize() - MessageFormatRecord.Crc_Size,
```

Comment:
you can't use `minfo.getSize()` here, if this is a Delete message info, then the size is 100. You should get the new MessageInfo from store.get method. This method returns StoreInfo, which has MessageInfo, which would be the size of the content.